src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java

```diff
@@ -1330,6 +1384,13 @@ public FormValidation doCheckSubnetsLaunchType(@QueryParameter("subnets") String
             return FormValidation.ok();
         }
 
+        public FormValidation doCheckOperatingSystemFamilyLaunchType(@QueryParameter("operatingSystemFamily") String operatingSystemFamily, @QueryParameter("launchType") String launchType) throws IOException, ServletException {
```

Comment:
Thanks @Stericson !
aptly noted, indeed, added validation check, 

although imho, for both operatingSystemFamily and cpuArchitecture fields, there is list selection forms on UI populated with valid values, also even in case of setting templates configuration from JCasC and omitting those values it will be set to defaults (LINUX, X86_64 ).

Reply:
@mslipets Yeah, I saw that you also handle empty/null values. But, just flagged more-so to match the convention/pattern that I saw you laying out. I thought about this for a while, and I did think, come to the conclusion, that it's clearer/better to force users that are making modifications to specify what they want while also providing backwards compatibility with existing configurations.

Though, now that I'm thinking of this, let me ask this. For users that have their clouds configured via code (CAC) will this cause those configurations. to break/fail due to a validation failure and require the user to update their CAC to explicitly specify these values? 

That scenario might make for the argument of not enforcing validation and just relying on your default if those value aren't provided.

Reply:
Thanks for sharing concerns @Stericson !

from my humble experience with current configuration in use, that's not causing any issues when fields missing from code,
it just falling back to defaults..

e.g. I've 2 agent templates for fargate, only for non default values need to be explicit..
![Screenshot from 2022-01-26 21-53-18](https://user-images.githubusercontent.com/53615853/151237235-1fe6effe-dcf4-4fd8-9bbb-e06e26dc2b60.png)

Reply:
Awesome, sounds good to me!